### ISSUE TITLE
Upgrade swarms to Flatcar 3139.2.2

### ISSUE THREAD

**Issue by cpswan** [atsign-foundation/at_server#739]
**Describe the solution you'd like**
Run the ripple upgrade scripts across staging and all production swarms.

Start with Canary.

**Comment by kumarnarendra701** [comment]
**Upgraded on Canary**

```
narendrakumar@Narendras-MacBook-Pro-2 flatcar_upgrade % ./canary_ripple_restart.sh
Target version is: 3139.2.2
Existing host keys found in /Users/narendrakumar/.ssh/google_compute_known_hosts
Processing canary-02
Warning: Permanently added 'compute.5048468463173161188' (ED25519) to the list of known hosts.
Found canary-02 at VERSION=3139.2.1
Draining node: canary-02
Existing host keys found in /Users/narendrakumar/.ssh/google_compute_known_hosts
canary-02.us-central1-b.c.secondaries.internal
I0618 16:44:35.120640 204236 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_UPDATED_NEED_REBOOT NEW_VERSION=3139.2.2
2 containers still running
Restarting canary-02
Stopping instance(s) canary-02...done.                                                                          
Updated [https://compute.googleapis.com/compute/v1/projects/secondaries/zones/us-central1-b/instances/canary-02].
Starting instance(s) canary-02...done.                                                                          
Updated [https://compute.googleapis.com/compute/v1/projects/secondaries/zones/us-central1-b/instances/canary-02].
Instance internal IP is 10.164.0.52
Instance external IP is 34.69.216.143
At 22:15:16 waiting 2m for canary-02 to be ready for login
Restarting Docker on canary-02
At 22:17:34 waiting 10m for CPU Load to return to normal on canary-02


Reactivating node: canary-02
Existing host keys found in /Users/narendrakumar/.ssh/google_compute_known_hosts
canary-02.us-central1-b.c.secondaries.internal
At 22:27:40 waiting 10m for Swarm to stabilise

Processing canary-03
Warning: Permanently added 'compute.147985497983209714' (ED25519) to the list of known hosts.
Found canary-03 at VERSION=3139.2.1
Draining node: canary-03
Existing host keys found in /Users/narendrakumar/.ssh/google_compute_known_hosts
canary-03.us-central1-c.c.secondaries.internal
I0618 17:07:58.004436 3730056 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_UPDATED_NEED_REBOOT NEW_VERSION=3139.2.2
57 containers still running
2 containers still running
Restarting canary-03
Stopping instance(s) canary-03...done.                                                                          
Updated [https://compute.googleapis.com/compute/v1/projects/secondaries/zones/us-central1-c/instances/canary-03].
Starting instance(s) canary-03...done.                                                                          
Updated [https://compute.googleapis.com/compute/v1/projects/secondaries/zones/us-central1-c/instances/canary-03].
Instance internal IP is 10.164.0.53
Instance external IP is 34.121.124.13
At 22:39:43 waiting 2m for canary-03 to be ready for login
Restarting Docker on canary-03
At 22:42:01 waiting 10m for CPU Load to return to normal on canary-03

Reactivating node: canary-03
Existing host keys found in /Users/narendrakumar/.ssh/google_compute_known_hosts
canary-03.us-central1-c.c.secondaries.internal
At 22:52:07 waiting 10m for Swarm to stabilise
```

**Comment by kumarnarendra701** [comment]
staging0001 upgraded to 3139.2.2

**Comment by kumarnarendra701** [comment]
swarm0001 upgraded to 3139.2.2

**Comment by kumarnarendra701** [comment]
swarm0002 upgraded to 3139.2.2

**Comment by kumarnarendra701** [comment]
swarm0003 upgraded to 3139.2.2

**Comment by kumarnarendra701** [comment]
@cpswan - Below are the errors during swarm0004-04 upgrade.

```
narendrakumar@Narendras-MacBook-Pro-2 flatcar_upgrade % ./swarm0004_ripple_restart.sh
Target version is: 3139.2.2
Existing host keys found in /Users/narendrakumar/.ssh/google_compute_known_hosts
Processing swarm0004-04
Found swarm0004-04 at VERSION=3139.2.1
Draining node: swarm0004-04
Existing host keys found in /Users/narendrakumar/.ssh/google_compute_known_hosts
swarm0004-04.us-central1-a.c.secondaries.internal


Updates are available for some Google Cloud CLI components.  To install them,
please run:
  $ gcloud components update

I0620 17:42:46.588865 1611010 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
3139.2.2 isn't ready yet - resetting update
I0620 17:42:51.787523 1611022 update_engine_client.cc:227] Setting Update Engine status to idle ...
I0620 17:42:51.789971 1611022 update_engine_client.cc:233] ResetStatus succeeded; to undo partition table changes run:
(D=$(rootdev -d) P=$(rootdev -s); cgpt p -i$(($(echo ${P#$D} | sed 's/^[^0-9]*//')-1)) $D;)
I0620 17:44:57.179913 1611093 update_engine_client.cc:241] Querying Update Engine status...
LAST_CHECKED_TIME=0
PROGRESS=0.000000
CURRENT_OP=UPDATE_STATUS_IDLE
NEW_VERSION=0.0.0
NEW_SIZE=0
I0620 17:45:02.315987 1611104 update_engine_client.cc:251] Initiating update check and install.
I0620 17:47:07.535307 1611165 update_engine_client.cc:251] Initiating update check and install.
I0620 17:47:12.634951 1611176 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_CHECKING_FOR_UPDATE NEW_VERSION=0.0.0
I0620 17:47:37.980500 1611207 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:48:03.289718 1611237 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:48:28.596634 1611268 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:48:53.958775 1611308 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:49:19.886703 1611335 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:49:44.974634 1611361 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:50:10.270202 1611387 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:50:35.477561 1611423 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:51:00.571365 1611469 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:51:25.947391 1611495 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:51:51.102555 1611528 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
I0620 17:52:16.882988 1611559 update_engine_client.cc:241] Querying Update Engine status...
CURRENT_OP=UPDATE_STATUS_IDLE NEW_VERSION=0.0.0
```

I have tried to debug and it's stuck in the while loop.
```
    while [[ "${STATUSLIST[2]}" != "CURRENT_OP=UPDATE_STATUS_UPDATED_NEED_REBOOT" ]]; do
      STATUS=$(gcloud compute ssh $GCPNODE \
        --command="sudo update_engine_client -status" -- -n)
      STATUSLIST=($STATUS)
      echo ${STATUSLIST[2]} ${STATUSLIST[3]}
      sleep 20
    done
```
Please share your thoughts.

**Comment by cpswan** [comment]
It seems swarm0004 has some residual problems from being built using patched images.

I've been able to force the update with:

```bash
sudo flatcar-update --to-version "3139.2.2" --force-flatcar-key
```

but also if I look at `/etc/flatcar/update.conf` I see:

```
GROUP=stable
REBOOT_STRATEGY="off"
SERVER=http://localhost:9090/update
```

That `SERVER` line shouldn't be there, and is what's preventing updates from being found. So I've deleted that.

As there's no load on that swarm yet I've rebooted all the workers, and they're now back up at 3139.2.2
